### PR TITLE
chore(open-ux-tools) Upgrade Dev Dependencies and fix Audit issues

### PR DESCRIPTION
**Pull request by devinea**
Fix Audit issues
- Upgrade core devDependencies
jest, eslint, babel, nx and related modules
- adapt tests to changes in jest 27 and 28
https://jestjs.io/docs/28.x/upgrading-to-jest28
https://jestjs.io/docs/upgrading-to-jest29

### CODE REVIEW

**Comment by ullasholla** [comment]
> Fix Audit issues
> 
>     * Upgrade core devDependencies
>       jest, typescript, eslint, babel, nx.
> 
>     * adapt tests to changes in jest 27 and 28
>       https://jestjs.io/docs/28.x/upgrading-to-jest28
>       https://jestjs.io/docs/upgrading-to-jest29

@devinea , could you change the PR description to leave `typescript` out? I don't see a change to TypeScript versions. Unless I'm mistaken or it's an oversight not changing it. Thanks

File: jest.base.js [hunk 1]
```diff
@@ -4,10 +4,7 @@ module.exports = {
     },
     collectCoverage: true,
     collectCoverageFrom: ['src/**/*.ts'],
-    coverageReporters: [
-        'text', 
-        ['lcov', { projectRoot: '../../' }]
-    ],
```
Comment (ullasholla) [inline jest.base.js:10]: Did automatic formatting change these lines? Ideally, pure reformatting should be left out of PRs

File: packages/backend-proxy-middleware/test/base/proxy.test.ts [hunk 2]
```diff
@@ -489,7 +489,7 @@ describe('proxy', () => {
             };
             const proxyOptions = await generateProxyMiddlewareOptions(backend, {}, logger);
             const debugSpy = jest.spyOn(logger, 'debug');
-            debugSpy.mockReset();
+            // debugSpy.mockReset();
```
Comment (ullasholla) [inline packages/backend-proxy-middleware/test/base/proxy.test.ts:492]: Delete the line instead?

File: packages/store/test/unit/services/backend-system.test.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ describe('BackendSystem service', () => {
     });
 
     beforeEach(() => {
-        jest.resetAllMocks();
+        // jest.resetAllMocks();
```
Comment (donal-tobin-sap) [inline packages/store/test/unit/services/backend-system.test.ts:15]: Not resetting the mocks between tests is probably a change in behavior and will probably break the tests, or the whole whole before each test is not required at all.

File: packages/store/test/unit/data-provider/backend-system.test.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ describe('Backend system data provider', () => {
 
     const logger = new ToolsLogger({ transports: [new NullTransport()] });
     beforeEach(() => {
-        jest.resetAllMocks();
+        // jest.resetAllMocks();
```
Comment (ullasholla) [inline packages/store/test/unit/data-provider/backend-system.test.ts:19]: Delete the line instead?

File: packages/store/test/unit/data-provider/system-migration.test.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ describe('System migration data provider', () => {
         readAll: jest.fn()
     };
     beforeEach(() => {
-        jest.resetAllMocks();
+        // jest.resetAllMocks();
```
Comment (ullasholla) [inline packages/store/test/unit/data-provider/system-migration.test.ts:18]: Delete the line instead?

File: packages/store/test/unit/data-provider/telemetry-setting.test.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ describe('TelemetrySetting data provider', () => {
         readAll: jest.fn()
     };
     beforeEach(() => {
-        jest.resetAllMocks();
+        // jest.resetAllMocks();
```
Comment (ullasholla) [inline packages/store/test/unit/data-provider/telemetry-setting.test.ts:18]: Delete the line instead?

File: packages/store/test/unit/services/backend-system.test.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ describe('BackendSystem service', () => {
...
     });
```
Comment (ullasholla) [inline packages/store/test/unit/services/backend-system.test.ts:16]: Remove it entirely?

File: packages/store/jest.config.js [hunk 8]
```diff
@@ -1,8 +1,11 @@
...
+config.setupFilesAfterEnv = ['jest-extended/all'];
+(config.transform = {
+    '^.+\\.tsx?$': [
+        'ts-jest',
+        {
+            tsconfig: 'test/tsconfig.json'
+        }
+    ]
+}),
+    (module.exports = config);
```
[the remark this answers is not in the document]
Comment (ullasholla) [reply packages/store/jest.config.js:11]: Question about the regex, the default regex is `'\\.[jt]sx+$'`. As we don't care for JS files, the pattern in this repo would be `'\\.tsx+$'`: https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object

Why do we use `'^.+\\.tsx?$'`?

**Review by ullasholla — CHANGES_REQUESTED** [review]
@devinea, thanks for the changes. Have a few questions, could you take a look please?

File: jest.base.js (hunk 1 above)
Comment (devinea) [reply jest.base.js:10]: Formatting the file when adding `snapshotFormat` below caused this.

File: docs/version-overrides.md [hunk 9]
```diff
@@ -2,55 +2,44 @@
...
 ## trim
-| Override:           | >=0.0.3 |
-|:--------------------| :-------------|
-|                     | |
-| **high**        | Regular Expression Denial of Service in trim  |
-| Package             | trim |
-| Vulnerable versions | <0.0.3 |
-| Patched versions    | >=0.0.3 |
-| More info           | https://github.com/advisories/GHSA-w5p7-h5w8-2hfq  |
+┌─────────────────────┬───────────────────────────────────────────────────┐
```
Comment (Klaus-Keller) [inline docs/version-overrides.md:5]: 1. Did you check whether the overrides for `trim`, `trim-newlines` and `glob-parent` are still required, even after update `storybook`? 
2. This looks now like this in markdown:
image

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]

File: packages/store/test/unit/services/backend-system.test.ts (hunk 3 above)
Comment (devinea) [reply packages/store/test/unit/services/backend-system.test.ts]: calling resetAllMocks breaks the tests. I think it maybe switching from jasmine to circus for the runner in jest.

File: packages/store/jest.config.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (devinea) [reply packages/store/jest.config.js]: I copied from tools-suite. `\\.tsx+$` causes errors for some reason

[the remark this answers is not in the document]
Comment (devinea) [reply packages/store/jest.config.js]: > I can't figure out why we need the parens and the , operator here. Help?
will remove. it is an copy and paste then format error

File: packages/backend-proxy-middleware/test/base/proxy.test.ts (hunk 2 above)
Comment (devinea) [reply packages/backend-proxy-middleware/test/base/proxy.test.ts:492]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/test/unit/services/backend-system.test.ts (hunk 3 above)
Comment (devinea) [reply packages/store/test/unit/services/backend-system.test.ts]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/test/unit/data-provider/backend-system.test.ts (hunk 4 above)
Comment (devinea) [reply packages/store/test/unit/data-provider/backend-system.test.ts]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/test/unit/data-provider/system-migration.test.ts (hunk 5 above)
Comment (devinea) [reply packages/store/test/unit/data-provider/system-migration.test.ts]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/test/unit/data-provider/telemetry-setting.test.ts (hunk 6 above)
Comment (devinea) [reply packages/store/test/unit/data-provider/telemetry-setting.test.ts]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/test/unit/services/backend-system.test.ts (hunk 7 above)
Comment (devinea) [reply packages/store/test/unit/services/backend-system.test.ts]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: packages/store/jest.config.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (devinea) [reply packages/store/jest.config.js]: Done in 3d67970ea55934680e91cd61e1b191bd65053b33

File: docs/version-overrides.md (hunk 9 above)
Comment (devinea) [reply docs/version-overrides.md]: > Did you check whether the overrides for trim, trim-newlines and glob-parent are still required, even after update storybook?

Yes they are still needed unfortunately. 

> This looks now like this in markdown:

Reverted in https://github.com/SAP/open-ux-tools/pull/891/commits/451a016a0d053b28afc0ca63f172884a71ef2bc1

Comment (Klaus-Keller) [reply docs/version-overrides.md]: ok, thanks @devinea

**Comment by devinea** [comment]
> could you change the PR description to leave `typescript` out? I don't see a change to TypeScript versions. Unless I'm mistaken or it's an oversight not changing it. Thanks

@ullasholla Thanks, updated.

**Review by Klaus-Keller — APPROVED** [review]
- changes in `environment-check` and `ui-components` look good. 
- did a local install, build, and test

Approved from my side

**Review by tobiasqueck — APPROVED** [review]
- `pnpm install && pnpm audit && pnpm build && pnpm test` all run without issues
- version changes all look reasonable
- changes in `store` tests due to the `jest` upgrade reviewed

File: packages/store/jest.config.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (tobiasqueck) [reply packages/store/jest.config.js]: Verified.

**Comment by devinea** [comment]
Thanks all

**Pull request merged**